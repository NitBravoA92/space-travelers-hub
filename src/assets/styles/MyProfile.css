.services-content {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  gap: 2rem;
}

.services-content .service {
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 0.8rem;
}

.services-content .service h2 {
  font-size: 2rem;
  color: #333;
}

.reservation-list {
  width: 100%;
  display: flex;
  flex-direction: column;
  gap: 0;
  border: 1px solid #ccc;
}

.reservation-item {
  padding: 0.7rem 0.8rem 1.1rem;
  border-bottom: 1px solid #ccc;
  font-size: 1rem;
  font-weight: 500;
  color: #333;
  transition: all 0.2s ease-out;
}

.reservation-item:hover {
  background-color: #f1f1f1;
}
