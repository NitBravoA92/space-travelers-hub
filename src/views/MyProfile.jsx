import Layout from './Layout';
import JoinedMissions from '../components/JoinedMissions';
import ReservedRockets from '../components/ReservedRockets';
import '../assets/styles/MyProfile.css';

const MyProfile = () => (
  <Layout>
    <section id="myProfile">
      <div className="container">
        <div className="services-content">
          <JoinedMissions />
          <ReservedRockets />
        </div>
      </div>
    </section>
  </Layout>
);

export default MyProfile;
