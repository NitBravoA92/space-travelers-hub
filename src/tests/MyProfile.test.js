import '@testing-library/jest-dom/extend-expect';
import { screen } from '@testing-library/react';
import { MemoryRouter } from 'react-router-dom';
import MyProfile from '../views/MyProfile';
import renderWithProviders from '../helpers/helper-for-test';

describe('The MyProfile component', () => {
  test('renders correctly into the DOM', () => {
    const { tree } = renderWithProviders(
      <MemoryRouter>
        <MyProfile />
      </MemoryRouter>,
    );

    expect(tree).toMatchSnapshot();
  });

  test("renders two <div> containers: 'My Missions' and 'My Rockets'", () => {
    renderWithProviders(
      <MemoryRouter>
        <MyProfile />
      </MemoryRouter>,
    );

    const renderMyMissionsContainer = document.querySelector('.my-missions');
    const renderMyRocketsContainer = document.querySelector('.my-rockets');
    const myMissionsTitle = screen.getByText(/My Missions/i);
    const myRocketsTitle = screen.getByText(/My Rockets/i);

    expect(renderMyMissionsContainer).toBeInTheDocument();
    expect(renderMyRocketsContainer).toBeInTheDocument();
    expect(myMissionsTitle).toBeInTheDocument();
    expect(myRocketsTitle).toBeInTheDocument();
  });

  test('should render the navigation menu with 3 links elements', () => {
    renderWithProviders(
      <MemoryRouter>
        <MyProfile />
      </MemoryRouter>,
    );

    const navigationMenu = document.querySelector('nav');
    const navLinks = document.querySelectorAll('nav ul li a');

    expect(navigationMenu).toBeInTheDocument();
    expect(navLinks).toHaveLength(3);
  });
});
