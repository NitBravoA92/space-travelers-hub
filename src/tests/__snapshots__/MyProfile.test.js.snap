// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The MyProfile component renders correctly into the DOM 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header>
        <div
          class="container"
        >
          <nav
            class="navbar"
          >
            <a
              class="logo"
              href="/"
            >
              <img
                alt="Space Travelers' Hub"
                class="logo-image"
                src="logo.png"
              />
              <span
                class="app-name"
              >
                Space Travelers' Hub
              </span>
            </a>
            <ul
              id="navigation"
            >
              <li
                class="nav-item"
              >
                <a
                  href="/rockets"
                >
                  Rockets
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  href="/missions"
                >
                  Missions
                </a>
              </li>
              <li
                class="nav-item"
              >
                <a
                  href="/my-profile"
                >
                  My Profile
                </a>
              </li>
            </ul>
          </nav>
        </div>
      </header>
      <main>
        <section
          id="myProfile"
        >
          <div
            class="container"
          >
            <div
              class="services-content"
            >
              <div
                class="my-missions service"
              >
                <h2>
                  My Missions
                </h2>
                <ul
                  class="reservation-list"
                  id="reserved-missions"
                />
              </div>
              <div
                class="my-rockets service"
              >
                <h2>
                  My Rockets
                </h2>
                <ul
                  class="reservation-list"
                  id="reserved-rockets"
                >
                  <li
                    class="reservation-item"
                  >
                    Falcon 1
                  </li>
                </ul>
              </div>
            </div>
          </div>
        </section>
      </main>
    </div>
  </body>,
  "container": <div>
    <header>
      <div
        class="container"
      >
        <nav
          class="navbar"
        >
          <a
            class="logo"
            href="/"
          >
            <img
              alt="Space Travelers' Hub"
              class="logo-image"
              src="logo.png"
            />
            <span
              class="app-name"
            >
              Space Travelers' Hub
            </span>
          </a>
          <ul
            id="navigation"
          >
            <li
              class="nav-item"
            >
              <a
                href="/rockets"
              >
                Rockets
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/missions"
              >
                Missions
              </a>
            </li>
            <li
              class="nav-item"
            >
              <a
                href="/my-profile"
              >
                My Profile
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </header>
    <main>
      <section
        id="myProfile"
      >
        <div
          class="container"
        >
          <div
            class="services-content"
          >
            <div
              class="my-missions service"
            >
              <h2>
                My Missions
              </h2>
              <ul
                class="reservation-list"
                id="reserved-missions"
              />
            </div>
            <div
              class="my-rockets service"
            >
              <h2>
                My Rockets
              </h2>
              <ul
                class="reservation-list"
                id="reserved-rockets"
              >
                <li
                  class="reservation-item"
                >
                  Falcon 1
                </li>
              </ul>
            </div>
          </div>
        </div>
      </section>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
